i18n-helpers/src/bin/mdbook-xgettext.rs

```diff
@@ -83,21 +104,148 @@ fn create_catalog(ctx: &RenderContext) -> anyhow::Result<Catalog> {
         add_message(&mut catalog, line, &source);
     }
 
+    let mut catalogs = HashMap::new();
+    let default_file_path = Path::new(input_path);
+    // The catalog from the summary data will exist in the single pot file for
+    // a depth of 0, or be presented in a separate `SUMMARY.pot` file for
+    // depths greater than 0.
+    if depth == 0 {
+        catalogs.insert(default_file_path.to_path_buf(), catalog);
+    } else {
+        catalogs.insert(summary_path, catalog);
+    }
+
     // Next, we add the chapter contents.
-    for item in ctx.book.iter() {
+    for item in &ctx.book.sections {
         if let BookItem::Chapter(chapter) = item {
             let path = match &chapter.path {
                 Some(path) => ctx.config.book.src.join(path),
                 None => continue,
             };
+
+            let rel_file_path = if depth == 0 {
+                default_file_path.to_path_buf()
+            } else {
+                catalogs.insert(path.clone(), Catalog::new(generate_catalog_metadata(&ctx)));
+                path.clone()
+            };
+
+            let catalog: &mut Catalog = catalogs.get_mut(&rel_file_path).with_context(|| {
```

Comment:
Just use `catalogs[&rel_file_path]` here: it is a program error if this fails, not a user error. So it's fine to panic and crash the program.

In general, you want to use `Result` for error handling when the errors are _expected_. This includes things like a file not existing or a TCP connection being dropped. Not being able to find something in a hashmap (which you just inserted!) is not expected — it's a bug in the program and the user cannot fix it by running the program again with different input.